lib/node/neutrino.js

```diff
@@ -0,0 +1,315 @@
+      maxFiles: this.config.uint('max-files'),
+      cacheSize: this.config.mb('cache-size'),
+      entryCache: this.config.uint('entry-cache'),
+      forceFlags: this.config.bool('force-flags'),
+      checkpoints: this.config.bool('checkpoints'),
+      bip91: this.config.bool('bip91'),
+      bip148: this.config.bool('bip148'),
+      spv: true,
+      location: path.join(this.config.prefix, 'neutrino')
+    });
+
+    this.filterIndexers.set(
+      'BASIC',
+      new FilterIndexer({
+        network: this.network,
+        logger: this.logger,
+        blocks: this.blocks,
+        chain: this.chain,
+        memory: this.config.bool('memory'),
+        prefix: this.config.str('index-prefix', this.config.prefix),
+        filterType: 'BASIC',
+        neutrino: true
+      })
+    );
+
+    this.pool = new Pool({
+      network: this.network,
+      logger: this.logger,
+      chain: this.chain,
+      prefix: this.config.prefix,
+      checkpoints: true,
+      filterIndexers: this.filterIndexers,
+      proxy: this.config.str('proxy'),
+      onion: this.config.bool('onion'),
+      upnp: this.config.bool('upnp'),
+      seeds: this.config.array('seeds'),
+      nodes: this.config.array('nodes'),
+      only: this.config.array('only'),
+      maxOutbound: this.config.uint('max-outbound'),
+      createSocket: this.config.func('create-socket'),
+      memory: this.memory,
+      selfish: true,
+      listen: false,
+      neutrino: this.neutrino,
+      spv: this.spv
+    });
+
+    this.rpc = new RPC(this);
+
+    this.http = new HTTP({
+      network: this.network,
+      logger: this.logger,
+      node: this,
+      prefix: this.config.prefix,
+      ssl: this.config.bool('ssl'),
+      keyFile: this.config.path('ssl-key'),
+      certFile: this.config.path('ssl-cert'),
+      host: this.config.str('http-host'),
+      port: this.config.uint('http-port'),
+      apiKey: this.config.str('api-key'),
+      noAuth: this.config.bool('no-auth'),
+      cors: this.config.bool('cors')
+    });
+
+    this.init();
+  }
+
+  /**
+   * Initialize the node.
+   * @private
+   */
+
+  init() {
+    // Bind to errors
+    this.chain.on('error', err => this.error(err));
+    this.pool.on('error', err => this.error(err));
+
+    if (this.http)
+      this.http.on('error', err => this.error(err));
+
+    this.chain.on('block', (block) => {
+      this.emit('block', block);
+    });
+
+    this.chain.on('connect', async (entry, block) => {
+      this.emit('connect', entry, block);
+    });
+
+    this.chain.on('disconnect', (entry, block) => {
+      this.emit('disconnect', entry, block);
+    });
+
+    this.chain.on('reorganize', (tip, competitor) => {
+      this.emit('reorganize', tip, competitor);
+    });
+
+    this.chain.on('reset', (tip) => {
+      this.emit('reset', tip);
+    });
+
+    this.pool.on('headers', async () => {
+      if (this.chain.height === 0)
+        return;
+      this.logger.info('Block Headers are fully synced');
```

Comment:
I don't think this is correct. The `headers` event is emitted by pool in `_handleHeaders()` **every time** we handle a headers packet, not when we are fully synced.

Reply:
I think what we should do, in `_handleHeaders()`, is once we are done processing new headers, we check is chain is synced (i.e. IBD is done). If chain is synced we immedeately call get filters. otherwise, wait for chain to emit `'full'` and trigger the first getfilters off of that.